File: tools/orchestrator/src/util/textUtil.ts
const STOP_WORDS = new Set([
  "the",
  "and",
  "for",
  "with",
  "you",
  "your",
  "our",
  "are",
  "will",
  "this",
  "that",
  "from",
  "into",
  "have",
  "has",
  "had",
  "their",
  "they",
  "them",
  "who",
  "what",
  "when",
  "where",
  "why",
  "how",
  "about",
  "all",
  "any",
  "can",
  "may",
  "able",
  "work",
  "working"
]);

export function tokenize(text: string): string[] {
  return text
    .toLowerCase()
    .replace(/[^a-z0-9]+/g, " ")
    .split(/\s+/)
    .map((t) => t.trim())
    .filter((t) => t.length >= 3)
    .filter((t) => !STOP_WORDS.has(t));
}

export function normalizeCompanyName(name: string): string {
  return name.replace(/\s+/g, " ").trim();
}

export function safeSlug(input: string): string {
  return input
    .toLowerCase()
    .replace(/[^a-z0-9]+/g, "")
    .slice(0, 40);
}

export function safeBasename(value: string): string {
  return value
    .trim()
    .toLowerCase()
    .replace(/[^a-z0-9]+/g, "_")
    .replace(/^_+|_+$/g, "")
    .slice(0, 48);
}

File: backend/src/resume_orchestrator/util/text_util.py
from __future__ import annotations

import re

_STOP_WORDS = frozenset(
    [
        "the", "and", "for", "with", "you", "your", "our", "are", "will",
        "this", "that", "from", "into", "have", "has", "had", "their",
        "they", "them", "who", "what", "when", "where", "why", "how",
        "about", "all", "any", "can", "may", "able", "work", "working",
    ]
)

_NON_ALNUM_RE = re.compile(r"[^a-z0-9]+")
_WHITESPACE_RE = re.compile(r"\s+")
_LEADING_TRAILING_UNDERSCORE_RE = re.compile(r"^_+|_+$")


def tokenize(text: str) -> list[str]:
    lowered = text.lower()
    spaced = _NON_ALNUM_RE.sub(" ", lowered)
    tokens = [t.strip() for t in spaced.split()]
    return [t for t in tokens if len(t) >= 3 and t not in _STOP_WORDS]


def normalize_company_name(name: str) -> str:
    return _WHITESPACE_RE.sub(" ", name).strip()


def safe_slug(value: str) -> str:
    return _NON_ALNUM_RE.sub("", value.lower())[:40]


def safe_basename(value: str) -> str:
    lowered = value.strip().lower()
    underscored = _NON_ALNUM_RE.sub("_", lowered)
    trimmed = _LEADING_TRAILING_UNDERSCORE_RE.sub("", underscored)
    return trimmed[:48]
